src/azure-cli-core/azure/cli/core/mock.py

```diff
@@ -17,16 +17,23 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
         from azure.cli.core.cloud import get_active_cloud
         from azure.cli.core.parser import AzCliCommandParser
         from azure.cli.core.util import random_string
-        from azure.cli.core._config import GLOBAL_CONFIG_DIR, ENV_VAR_PREFIX
+        from azure.cli.core._config import GLOBAL_CONFIG_DIR, ENV_VAR_PREFIX, ENV_VAR_TEST_LIVE
         from azure.cli.core._help import AzCliHelp
         from azure.cli.core._output import AzOutputProducer
 
         from knack.completion import ARGCOMPLETE_ENV_NAME
 
+        # random_config_dir only takes effects in playback mode
+        if random_config_dir and os.getenv(ENV_VAR_TEST_LIVE, '').lower() != 'true':
+            config_dir = os.path.join(GLOBAL_CONFIG_DIR, 'dummy_cli_config_dir', random_string())
+            self.use_random_config_dir = True
+        else:
+            config_dir = GLOBAL_CONFIG_DIR
+            self.use_random_config_dir = False
```

Comment:
Won't this trigger failure when executed concurrently in live mode?

Reply:
That's a valid point. I need to copy login credentials then.